### PR TITLE
Editor controller setlang

### PR DESCRIPTION
**Pull request by pelikhan**
Support setting editor language from editor controller
- [x] moving cookie lang stuff to BrowserUtils

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -3951,8 +3951,7 @@ document.addEventListener("DOMContentLoaded", () => {
                 live,
                 force)
                 .then(() => {
-                    if (pxt.Util.isLocaleEnabled(useLang)) {
-                        lang.setCookieLang(useLang);
+                    if (pxt.BrowserUtils.setCookieLang(useLang)) {
```
Comment (abchatra) [inline webapp/src/app.tsx]: This is incorrect. We need to check isLocaleEnabled which checks appTheme.availableLocales before enabling the locale for a particular editor

File: pxteditor/editorcontroller.ts [hunk 2]
```diff
@@ -413,6 +419,16 @@ namespace pxt.editor {
                                             }
                                         });
                                 }
+                                case "setlang": {
+                                    const langmsg = data as EditorMessageSetLangRequest;
+                                    const curr = pxt.BrowserUtils.getCookieLang();
+                                    if (pxt.BrowserUtils.setCookieLang(langmsg.lang)
```
[the remark this answers is not in the document]
Comment (abchatra) [reply pxteditor/editorcontroller.ts]: Setting cookie has implications on the cookie banner.

File: webapp/src/app.tsx (hunk 1 above)
Comment (pelikhan) [reply webapp/src/app.tsx]: the check happens in setCookieLang

File: pxteditor/editorcontroller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pelikhan) [reply pxteditor/editorcontroller.ts]: how is the editorcontroller supposed to set the language?

File: webapp/src/app.tsx (hunk 1 above)
Comment (abchatra) [reply webapp/src/app.tsx]: That is not the same check. It checks all the languages. Each editor has a specific language checked.

File: pxteditor/editorcontroller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (abchatra) [reply pxteditor/editorcontroller.ts]: why does it need to? Controller always sets the langauge required.

**Pull request closed without merging**